registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/ABISHandlerUtil.java

```diff
@@ -235,4 +275,81 @@ public List<String> getUniqueRegIds(List<String> matchedRegistrationIds, String
 
 	}
 
+	public void validateBiometricRecord(BiometricRecord biometricRecord, List<String> modalities)
+			throws BiometricRecordValidationException, JsonParseException, JsonMappingException, IOException {
+		if (modalities == null || modalities.isEmpty()) {
+			throw new BiometricRecordValidationException(PlatformErrorMessages.RPR_DATASHARE_MODALITIES_EMPTY.getCode(),
+					PlatformErrorMessages.RPR_DATASHARE_MODALITIES_EMPTY.getMessage());
+		}
+		if (biometricRecord == null || biometricRecord.getSegments() == null
+				|| biometricRecord.getSegments().isEmpty()) {
+			throw new BiometricRecordValidationException(
+					PlatformErrorMessages.RPR_NO_BIOMETRICS_FOUND_WITH_DATASHARE.getCode(),
+					PlatformErrorMessages.RPR_NO_BIOMETRICS_FOUND_WITH_DATASHARE.getMessage());
+		}
+
+		for (String segment : modalities) {
+			Optional<BIR> optionalBIR = Optional.empty();
+			if (segment.equalsIgnoreCase("Face")) {
+				optionalBIR = biometricRecord.getSegments().stream().filter(bir -> bir.getBdbInfo().getType() != null
+						&& bir.getBdbInfo().getType().get(0).equals(BiometricType.FACE)).findFirst();
+			} else {
+				String[] segmentArray = segment.split(" ");
+				optionalBIR = biometricRecord.getSegments().stream().filter(bir -> bir.getBdbInfo().getSubtype() != null
+						&& bir.getBdbInfo().getSubtype().size() == segmentArray.length
+								? (bir.getBdbInfo().getSubtype().get(0).equalsIgnoreCase(segmentArray[0])
+										&& (segmentArray.length == 2
+												? bir.getBdbInfo().getSubtype().get(1).equalsIgnoreCase(segmentArray[1])
+												: true))
+								: false)
+						.findFirst();
+			}
+			if (optionalBIR.isPresent()) {
+				BIR bir = optionalBIR.get();
+				if (bir.getBdb() != null) {
+					return;
+				}
+			}
+		}
+
+		throw new BiometricRecordValidationException(
+				PlatformErrorMessages.RPR_NO_BIOMETRIC_MATCH_WTIH_DATASAHRE.getCode(),
+				PlatformErrorMessages.RPR_NO_BIOMETRIC_MATCH_WTIH_DATASAHRE.getMessage());
+	}
+
+	public Map<String, List<String>> createTypeSubtypeMapping() throws ApisResourceAccessException, DataShareException,

```

Comment:
can we rename this method, it does not give any details about what it is doing

Reply:
fixed